assets/queries/k8s/containers_run_with_low_uid/query.rego

```diff
@@ -1,39 +1,63 @@
 package Cx
 
 import data.generic.common as common_lib
+import data.generic.k8s as k8sLib
 
-CxPolicy[result] {
-	doc := input.document[i]
-
-	[path, value] = walk(doc)
-	securityContext := value.securityContext
+types := {"initContainers", "containers"}
 
-	to_number(securityContext.runAsUser) < 10000
+# container defines runAsUser
+checkUser(specInfo, container, containerType, document, metadata) = result {
+	uid := container.securityContext.runAsUser
+	to_number(uid) < 10000
 
 	result := {
-		"documentId": doc.id,
-		"searchKey": sprintf("%s.securityContext.runAsUser=%d", [common_lib.concat_path(path), securityContext.runAsUser]),
+		"documentId": document.id,
+		"searchKey": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.runAsUser=%d", [metadata.name, specInfo.path, containerType, container.name, uid]),
 		"issueType": "IncorrectValue",
-		"keyExpectedValue": sprintf("%s.securityContext.runAsUser should not be a low UID", [common_lib.concat_path(path)]),
-		"keyActualValue": sprintf("%s.securityContext.runAsUser is a low UID", [common_lib.concat_path(path)]),
-		"searchLine": common_lib.build_search_line(path, ["securityContext", "runAsUser"]),
+		"keyExpectedValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.runAsUser should be set to a UID >= 10000", [metadata.name, specInfo.path, containerType, container.name]),
+		"keyActualValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.runAsUser is set to a low UID", [metadata.name, specInfo.path, containerType, container.name]),
 	}
 }
 
-CxPolicy[result] {
-	doc := input.document[i]
+# pod defines runAsUser and container inherits this setting
+checkUser(specInfo, container, containerType, document, metadata) = result {
+	containerCtx := object.get(container, "securityContext", {})
+	not common_lib.valid_key(containerCtx, "runAsUser")
```

Comment:
```suggestion
	not common_lib.valid_key(container.securityContext, "runAsUser")
```

Reply:
The desired condition here is: `runAsUser` is not defined in a _potentially_ existing `container.securityContext` object. 

The `object.get(.., {})` check ensures that the securityContext can exist but doesn't necessarily have to. With the check you suggested the rule would evaluate to _false_, in case no securityContext is defined on container level. However, this can be perfectly legit, since `runAsUser` may instead be defined on pod level.